stl/inc/optional

```diff
@@ -239,19 +239,23 @@ public:
 
     template <class _Ty2>
     using _AllowDirectConversion = bool_constant<conjunction_v<negation<is_same<_Remove_cvref_t<_Ty2>, optional>>,

```

Comment:
Pre-existing, but why is this doing a `bool_constant<conjunction_v` rather than just `conjunction`; no change requested.

Reply:
I think there's some attempt here to improve throughput by "smashing" a complicated type to either `true_type` or `false_type` but looking at this again, I am unsure whether we are actually achieving anything here. In cases below (e.g. `_AllowUnwrappingAssignment`) we are introducing a new `struct` instead of simply a `using`, and there appears to be no short-circuit benefit to doing so. I'll record a todo on my list, but I would prefer to avoid touching this existing code in `<optional>` until Casey returns as there is no urgent need.